tests/testthat/test-data-set.R

```diff
@@ -121,6 +121,35 @@ test_that("it can update the unit of the yValues and this does not change the re
   expect_equal(dataSet$yErrorUnit, ospUnits$`Concentration [mass]`$`mg/l`)
 })
 
+test_that("it can update the dimension of the yValues and this does not change the returned value of the error", {
+  dataSet <- DataSet$new(name = dataSetName)
+  expectedYErrorValues <- c(0.001, 0.001, 0.1, 1)
+  dataSet$setValues(
+    xValues = c(1, 2, 3, 4),
+    yValues = c(0, 0.1, 0.6, 10),
+    yErrorValues = expectedYErrorValues
+  )
+
+  dataSet$yErrorUnit <- ospUnits$`Concentration [mass]`$`pg/l`
+  dataSet$yDimension <- ospDimensions$Amount
+
+  expect_equal(dataSet$yErrorValues, expectedYErrorValues, tolerance)
+})
+
+test_that("it can update the unit of the yErrorValues and this does not change the returned value", {
+  dataSet <- DataSet$new(name = dataSetName)
+  expectedYErrorValues <- c(0.001, 0.001, 0.1, 1)
+  dataSet$setValues(
+    xValues = c(1, 2, 3, 4),
+    yValues = c(0, 0.1, 0.6, 10),
+    yErrorValues = expectedYErrorValues
+  )
+  # Change units of y error values - actual values are converted
+  dataSet$yErrorUnit <- ospUnits$`Concentration [mass]`$`pg/l`
+
+  expect_equal(dataSet$yErrorValues, expectedYErrorValues, tolerance)
+})
+
```

Comment:
Can't these be collapsed to form a single test, so that there is no repetition?

```r
test_that("it can update the unit and dimension of the yValues and this does not change the returned value of the error", {
  dataSet <- DataSet$new(name = dataSetName)
  expectedYErrorValues <- c(0.001, 0.001, 0.1, 1)
  dataSet$setValues(
    xValues = c(1, 2, 3, 4),
    yValues = c(0, 0.1, 0.6, 10),
    yErrorValues = expectedYErrorValues
  )

  dataSet$yErrorUnit <- ospUnits$`Concentration [mass]`$`pg/l`
  expect_equal(dataSet$yErrorValues, expectedYErrorValues, tolerance)
    
  dataSet$yDimension <- ospDimensions$Amount
  expect_equal(dataSet$yErrorValues, expectedYErrorValues, tolerance)
})
```

Reply:
Yes, did this, and also fixed an introduced bug.